File: backend/3PPASSWORD.BACKEND/Models/Entities/Credential.cs
namespace _3PPASSWORD.BACKEND.Models.Entities
{
    public class Credential
    {
        public Guid Id { get; set; }
        public string SystemName { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public string EncryptedPassword { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

    }
}

File: frontend/3PPASSWORD.FRONTEND/src/types/credential.ts
export interface Credential {
    id: string;
    systemName: string;
    username: string;
    url: string;
    notes: string;
    createdAt: string;
    updateAt: string;
}

export interface CreateCredencialRequest {
    systemName: string;
    username: string;
    url: string;
    password: string;
    repeatPassword: string;
    notes: string;
}

export interface CredentialPassword{
    id: string;
    systemName: string;
    password: string;
}